action/protocol/staking/ethabi/stake_base.go

```diff
@@ -0,0 +1,162 @@
+package ethabi
+
+import (
+	"encoding/hex"
+	"errors"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/iotexproject/iotex-core/pkg/util/addrutil"
+	"github.com/iotexproject/iotex-proto/golang/iotexapi"
+	"github.com/iotexproject/iotex-proto/golang/iotextypes"
+)
+
+var (
+	errInvalidCallData  = errors.New("invalid call binary data")
+	errInvalidCallSig   = errors.New("invalid call sig")
+	errConvertBigNumber = errors.New("convert big number error")
+	errDecodeFailure    = errors.New("decode data error")
+)
+
+type (
+	// Parameters state request parameters
+	Parameters struct {
+		MethodName []byte
+		Arguments  [][]byte
+	}
+
+	// StateContext context for ReadState
+	StateContext interface {
+		Parameters() *Parameters
+		EncodeToEth(*iotexapi.ReadStateResponse) (string, error)
+	}
```

Comment:
it seems that these structs/interfaces are used in multiple protocols. Thus, we may unify them into, say, action/protocol/